src/js/views/Project/Create/Create.jsx

```diff
@@ -1,156 +1,227 @@
-    (globalState.integrations.gitlab.enabled &&
-      user.integrations.includes('gitlab')) ||
-    globalState.integrations.grafana.enabled ||
-    globalState.integrations.sentry.enabled ||
-    globalState.integrations.sonarqube.enabled
-  const [sidebarLinks, setSidebarLinks] = useState(buildSidebarLinks())
+  const [globalState] = useContext(Context)
+  const [errorMessage, setErrorMessage] = useState(null)
+  const [fieldValues, setFieldValues] = useState(EMPTY_FIELDS)
+  const [fieldErrors, setFieldErrors] = useState(EMPTY_FIELDS)
+  const [isReady, setIsReady] = useState(false)
+  const [isSaving, setIsSaving] = useState(false)
+  const [automations, setAutomations] = useState([])
+  const [selectedAutomations, setSelectedAutomations] = useState([])
 
-  function buildSidebarLinks() {
-    const links = [attributesLink]
-    if (
-      localState.attributes.environments !== null &&
-      localState.attributes.environments.length > 0
-    )
-      links.push(urlsLink)
-    if (automationsEnabled) links.push(automationsLink)
-    links.push(linksLink)
-    return links
-  }
+  useEffect(async () => {
+    /* fetch automations available for selected project type */
+    if (fieldValues.project_type_id) {
+      const url = new URL('/ui/available-automations', globalState.baseURL)
+      url.searchParams.append('project_type_id', fieldValues.project_type_id)
+      await httpGet(
+        globalState.fetch,
+        url,
+        ({ data }) => {
+          setAutomations(
+            data.map(({ automation_name, integration_name, automation_slug }) =>
+              Automation({
+                automationName: automation_name,
+                integrationName: integration_name,
+                automationSlug: automation_slug
+              })
+            )
+          )
+        },
+        ({ error }) => {
+          setErrorMessage(error)
+        }
+      )
+    }
+  }, [fieldValues.project_type_id])
 
   useEffect(() => {
-    globalDispatch({
-      type: 'SET_CURRENT_PAGE',
-      payload: {
-        title: 'projects.newProject',
-        url: new URL('/ui/projects/create', globalState.baseURL)
+    /* remove any previously selected automations that are no longer available */
+    const automationSlugs = new Set(automations.map((a) => a.automationSlug))
+    setSelectedAutomations((prevState) =>
+      prevState.filter((slug) => automationSlugs.has(slug))
+    )
+  }, [automations])
+
+  const onChange = (key, value) => {
+    /* update fieldValues property named `key` */
+    if (fieldValues[key] !== value) {
+      const newValues = { ...fieldValues, [key]: value }
+      const result = validate(newValues, jsonSchema)
+      const errors = {}
+      if (result.errors.length > 0) {
+        setIsReady(false)
+        result.errors.map((err) => {
+          err.path.map((field) => {
+            if (newValues[field] !== null) {
+              errors[field] = err.message
+            }
+          })
+        })
```

Comment:
If these are arrays, this should use `forEach` instead, `map` is generating an unnecessary list.

```javascript
results.errors.forEach((err) => {
  err.path.forEach((field) => {
    if (newValues[field] !== null) {
      errors[field] = err.message
    }
  })
})
```